packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -191,22 +187,82 @@ export interface ContainerRuntimeMessage {
     contents: any;
     type: ContainerMessageType;
 }
+export interface ISummaryConfigurationBaseSettings {
+    /**
+     *  Delay before first attempt to spawn summarizing container.
+     * */
+    initialSummarizerDelayMs: number;
+
+    /**
+     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
+     * This defaults to false (disabled) and must be explicitly set to true to enable.
+     */
+    summarizerClientElection: boolean;
+
+    /**
+     * Defines the maximum allowed time to wait for a pending summary ack.
+     * The maximum amount of time client will wait for a summarize is the minimum of
+     * maxSummarizeAckWaitTime ( currently 10 minutes (10 * 60 * 1000) and maxAckWaitTime.
+     */
+    maxAckWaitTime: number;
+    /**
+     * Defines the maximum number of Ops in between Summaries that can be
+     * allowed before forcibly electing a new summarizer client.
+     */
+    maxOpsSinceLastSummary: number;
+}
+
+export interface ISummaryConfigurationHeuristicSettings extends ISummaryConfigurationBaseSettings {
+    /**
+     * Defines the maximum allowed time in between summarizations.
+     */
+     idleTime: number;
+    /**
+     * Defines the maximum allowed time, since the last received Ack,  before running the summary
+     * with reason maxTime.
+     */
+    maxTime: number;
+    /**
+     * Defines the maximum number of Ops, since the last received Ack, that can be allowed
+     * before running the summary with reason maxOps.
+     */
+     maxOps: number;
+    /**
+     * Defnines the minimum number of Ops, since the last received Ack, that can be allowed
+     * before running the last summary.
+     */
+    minOpsForAttemptOnClose: number;
+}
+
+export type ISummaryConfiguration =
+{
+    state: "disabled";
+}
+| ({ state: "disableHeuristics"; } & ISummaryConfigurationBaseSettings)
+| ({ state: "enabled";} & ISummaryConfigurationHeuristicSettings);
 
 // Consider idle 5s of no activity. And snapshot if a minute has gone by with no snapshot.
-const IdleDetectionTime = 5000;
+export const IdleDetectionTime = 5000;
 
-const DefaultSummaryConfiguration: ISummaryConfiguration = {
+export const DefaultSummaryConfiguration: ISummaryConfiguration = {
+    state: "enabled",
     idleTime: IdleDetectionTime * 3,
 
     maxTime: IdleDetectionTime * 12,
-
     // Summarize if 1000 ops received since last snapshot.
     maxOps: 1000,
 
-    // Wait 10 minutes for summary ack
+    minOpsForAttemptOnClose: 50,
+    // Wait 2 minutes for summary ack
```

Comment:
>2 [](http://example.com/codeflow?start=12&length=1)

10

Reply:
Not sure what this comment means but the maxAckWaitTime is still 10 mins :-).

Reply:
I'd remove comment (mentioning number) and format it as maxAckWaitTime: 6* 10 *1000; // 6 min
A bit more readable :)